src/utils/filter-utils.js

```diff
@@ -304,12 +368,74 @@ export function getFieldDomain(data, field) {
   }
 }
 
+export const getPolygonFilterFunctor = (layer, filter) => {
+  const getPosition = layer.getPositionAccessor();
+
+  switch (layer.type) {
+    case LAYER_TYPES.arc:
+    case LAYER_TYPES.line:
+      return data => {
+        const pos = getPosition({data});
+        return pos.every(Number.isFinite) && [
+          [pos[0], pos[1]],
+          [pos[3], pos[4]]
+        ].every(point => isInPolygon(point, filter.value));
+      };
+    default:
+      return data => {
```

Comment:
not all getPosition from layer returns point, i think for other layer type we should simply return `data => true`